Library/ViewModels/ShareViewModelTests.swift

```diff
@@ -23,6 +23,17 @@ internal final class ShareViewModelTests: TestCase {
     self.vm.outputs.showShareSheet.observe(self.showShareSheet.observer)
   }
 
+  func testShowShareSheet_Discovery() {
+    let project = .template |> Project.lens.id .~ 30
+    let newProject = .template |> Project.lens.id .~ 55
+
+    self.vm.inputs.configureWith(shareContext: .discovery(project))
+    self.vm.inputs.shareButtonTapped()
+    self.vm.inputs.configureWith(shareContext: .discovery(newProject))
+
+    self.showShareSheet.assertValueCount(1)
+  }
```

Comment:
you could also assert that the new share context tracking string you added is being passed here.